File: Modules/UtilModule.cs
using Discord.Commands;
using Discord.WebSocket;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DiscordBot.Modules
{
    public class UtilModule : ModuleBase<SocketCommandContext>
    {
        private readonly CommandService _commandService;
        private readonly DiscordSocketClient _discord;

        public UtilModule(CommandService commandService, DiscordSocketClient discord)
        {
            _commandService = commandService;
            _discord = discord;
        }

        string BuildCommandString(CommandInfo command)
        {
            var aliases = string.Join(' ', command.Aliases
                .Where(a => a != command.Name)
                .Select(a => $"{string.Join(", ", a)}"));
            var aliasString = command.Aliases.Any(a => a != command.Name) ? $" (Aliases: {aliases})" : "";
            var parameters = string.Join(' ', command.Parameters.Select(p => $"<{p.Name}>"));
            return $"{command.Name}{aliasString}{(command.Parameters.Any() ? " " : "")}{parameters}";
        }

        [Command("help")]
        [Summary("Shows this command list.")]
        public Task Help()
        {
            var commands = _commandService.Commands
                .Where(c => !string.IsNullOrEmpty(c.Summary) && !c.Preconditions.Any(p => p.Group == "Permission"));

            var embedBuilder = new Discord.EmbedBuilder();
            embedBuilder.WithTitle("Commands");
            foreach (var command in commands)
            {
                var embedFieldBuilder = new Discord.EmbedFieldBuilder()
                    .WithIsInline(false)
                    .WithName(BuildCommandString(command))
                    .WithValue(command.Summary);
                embedBuilder.AddField(embedFieldBuilder);
            }

            var isAdmin = Context.User.Id == Context.Guild.OwnerId;
            if (isAdmin)
            {
                embedBuilder.WithTitle("Admin Commands");
                var adminCommands = _commandService.Commands
                    .Where(c => !string.IsNullOrEmpty(c.Summary)
                             && c.Preconditions.Any(p => p.Group == "Permission"));
                foreach (var command in adminCommands)
                {
                    var embedFieldBuilder = new Discord.EmbedFieldBuilder()
                        .WithIsInline(false)
                        .WithName(BuildCommandString(command))
                        .WithValue(command.Summary);
                    embedBuilder.AddField(embedFieldBuilder);
                }
            }

            embedBuilder.WithDescription("To use the bot, tag it and specify one of the commands shown below.  "
                                       + "Replace the parts of commands surrounded by <> with your own text.\n"
                                       + "Example: " + _discord.CurrentUser.Mention + " set 2/31/2020")
                .WithFooter("Source code can be found at https://github.com/travv0/soberbot\n"
                          + "Please report any bugs at https://github.com/travv0/soberbot/issues");

            return ReplyAsync(null, false, embedBuilder.Build());
        }
    }
}

File: Modules/UtilModule.fs
namespace SoberBot.Modules

open Discord.Commands
open System
open System.Threading.Tasks

type UtilModule() =
    inherit ModuleBase<SocketCommandContext>()

    let buildCommandString (command: CommandInfo) =
        let aliases =
            String.Join(
                ' ',
                command.Aliases
                |> Seq.filter ((<>) command.Name)
                |> Seq.map (fun a -> sprintf "%s" (String.Join(", ", a)))
            )

        let aliasString =
            if command.Aliases |> Seq.contains command.Name then
                $" (Aliases: {aliases})"
            else
                ""

        let parameters =
            String.Join(
                ' ',
                command.Parameters
                |> Seq.map (fun p -> $"<{p.Name}>")
            )

        sprintf
            "%s%s%s%s"
            command.Name
            aliasString
            (if command.Parameters.Count > 0 then
                 " "
             else
                 "")
            parameters

    [<Command("help"); Summary("Shows this command list.")>]
    member this.Help(): Task =
        let commands =
            Services.commands.Commands
            |> Seq.filter
                (fun c ->
                    not (String.IsNullOrEmpty(c.Summary))
                    && not (
                        c.Preconditions
                        |> Seq.exists (fun p -> p.Group = "Permission")
                    ))

        let embedBuilder =
            Discord.EmbedBuilder().WithTitle("Commands")

        for command in commands do
            let embedFieldBuilder =
                Discord
                    .EmbedFieldBuilder()
                    .WithIsInline(false)
                    .WithName(buildCommandString command)
                    .WithValue(command.Summary)

            embedBuilder.AddField(embedFieldBuilder) |> ignore


        embedBuilder
            .WithDescription("To use the bot, tag it and specify one of the commands shown below.  "
                             + "Replace the parts of commands surrounded by <> with your own text.\n"
                             + "**Example:** "
                             + Services.discord.CurrentUser.Mention
                             + " set 3/31/2020")
            .WithFooter(
                "Source code can be found at https://github.com/travv0/soberbot\n"
                + "Please report any bugs at https://github.com/travv0/soberbot/issues"
            )
        |> ignore

        this.ReplyAsync(null, false, embedBuilder.Build())
        |> Async.AwaitTask
        |> Async.Ignore
        |> Async.RunSynchronously

        let isAdmin =
            this.Context.User.Id = this.Context.Guild.OwnerId

        if isAdmin then
            let adminEmbedBuilder =
                Discord.EmbedBuilder().WithTitle("Admin Commands")

            let adminCommands =
                Services.commands.Commands
                |> Seq.filter
                    (fun c ->
                        not (String.IsNullOrEmpty(c.Summary))
                        && c.Preconditions
                           |> Seq.exists (fun p -> p.Group = "Permission"))

            for command in adminCommands do
                let embedFieldBuilder =
                    Discord
                        .EmbedFieldBuilder()
                        .WithIsInline(false)
                        .WithName(buildCommandString command)
                        .WithValue(command.Summary)

                adminEmbedBuilder.AddField(embedFieldBuilder)
                |> ignore

            this.ReplyAsync(null, false, adminEmbedBuilder.Build()) :> Task
        else
            Task.CompletedTask
